src/hip_memory.cpp

```diff
@@ -410,6 +410,113 @@ extern void getChannelOrderAndType(const hipChannelFormatDesc& desc,
                             hsa_ext_image_channel_order_t& channelOrder,
                             hsa_ext_image_channel_type_t& channelType);
```

Comment:
Use pointers (rather than references) for variables that are to be written so the interface is clear.